pkg/ast/ast_type.go

```diff
@@ -27,10 +27,24 @@ type Type struct {
 	OfType   int
 }
 
+func (d *Document) BaseTypeNameBytes(ref int) ByteSlice {
+	if kind := d.Types[ref].TypeKind; kind == TypeKindNonNull || kind == TypeKindList {
+		return d.TypeNameBytes(d.Types[ref].OfType)
```

Comment:
So this can be infinitely nested. You know you're at the root once OfType is -1.

Reply:
Oh i see, i did not know that, thanks for pointing that!